### PR TITLE
Add removeAllGameBuildings / restoreAllGameBuildings API

### PR DESCRIPTION
**Pull request by TheNormalnij**
After #3274 

removeAllGameBuildings frees the building pool completely. This allows creating 13000 buildings without SA world.

```lua
bool removeAllGameBuildings()
bool restoreAllGameBuildings()
```

### CODE REVIEW

File: Client/game_sa/CIplStoreSA.cpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+ *****************************************************************************/
+
+#include "StdInc.h"
+
+#include "CIplStoreSA.h"
+#include "CQuadTreeNodeSA.h"
+
+static auto gIplQuadTree = (CQuadTreeNodesSAInterface<CIplSAInterface>**)0x8E3FAC;
+
+CIplStoreSA::CIplStoreSA() : m_isStreamingEnbabled(true), m_ppIplPoolInterface((CPoolSAInterface<CIplSAInterface>**)0x8E3FB0)
```
Comment (tederis) [inline Client/game_sa/CIplStoreSA.cpp]: `m_isStreamingEnbabled` misprint

File: Client/game_sa/CBuildingsPoolSA.cpp [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+inline bool CBuildingsPoolSA::AddBuildingToPool(CClientBuilding* pClientBuilding, CBuildingSA* pBuilding)
+{
+    // Grab the new object interface
+    CBuildingSAInterface* pInterface = pBuilding->GetBuildingInterface();
+
+    if (!pInterface)
+        return false;
+
+    uint32_t dwElementIndexInPool = (*m_ppBuildingPoolInterface)->GetObjectIndexSafe(pInterface);
+    if (dwElementIndexInPool >= -1)
```
Comment (tederis) [inline Client/game_sa/CBuildingsPoolSA.cpp]: `dwElementIndexInPool` cannot be bigger than -1(0xffffffff). It's better to leave it `dwElementIndexInPool == -1`. Though, I personally prefer `dwElementIndexInPool == UINT_MAX` or `dwElementIndexInPool == std::numeric_limits<uint32_t>().max()`

Comment (tederis) [reply Client/game_sa/CBuildingsPoolSA.cpp]: The same at line 93.

**Comment by TheNormalnij** [comment]
I forgot to delete buildings elements too

**Comment by TheNormalnij** [comment]
Fixed

**Comment by Dutchman101** [comment]
Please resolve conflicts
@TheNormalnij

**Comment by Dutchman101** [comment]
It passed 1 code review, the rest can be tested in upcoming nightly (as soon build server is fixed) along with other related merged PR's

**Pull request merged**